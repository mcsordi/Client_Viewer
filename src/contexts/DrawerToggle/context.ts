import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
