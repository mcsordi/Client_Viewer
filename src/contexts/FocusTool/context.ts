import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
