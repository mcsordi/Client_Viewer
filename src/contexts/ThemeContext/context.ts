import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

