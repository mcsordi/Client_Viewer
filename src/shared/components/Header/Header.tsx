import React, { useContext } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context.ts';

export const Header: React.FC<{ header: string }> = ({ header }) => {
  const { light } = useContext(themeContext);
  return (
    <h1
      className={`${light ? `text-black` : `text-white`}  text-2xl font-roboto sm:text-3xl md:text-4xl`}
    >
      {header}
    </h1>
  );
};
