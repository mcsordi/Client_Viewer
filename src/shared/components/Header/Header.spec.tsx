import { render, screen } from '@testing-library/react';
import { Header } from './Header';
import React from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context.ts';

describe('<Header/>', () => {
  const changeTheme = () => {};
  it('should render a Header component', () => {
    render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <Header header="Titulo" />
      </themeContext.Provider>,
    );
    const title = screen.getByRole('heading', { name: 'Titulo' });
    expect(title).toBeInTheDocument();
  });
  it('should render a Header component in the dark theme', () => {
    render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <Header header="Titulo" />
      </themeContext.Provider>,
    );
    const title = screen.getByRole('heading', { name: 'Titulo' });
    expect(title).toHaveClass('text-white');
  });
  it('should render a Header component in the light theme', () => {
    render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <Header header="Titulo" />
      </themeContext.Provider>,
    );
    const title = screen.getByRole('heading', { name: 'Titulo' });
    expect(title).toHaveClass('text-black');
    expect(title).not.toHaveClass('text-white');
  });
  it('should match a snapshot', () => {
    render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <Header header="Titulo" />
      </themeContext.Provider>,
    );
    const title = screen.getByRole('heading', { name: 'Titulo' });
    expect(title).toMatchSnapshot();
  });
});
