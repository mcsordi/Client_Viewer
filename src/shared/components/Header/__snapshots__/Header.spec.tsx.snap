// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match a snapshot 1`] = `
<h1
  class="text-black  text-2xl font-roboto sm:text-3xl md:text-4xl"
>
  Titulo
</h1>
`;
